grakel_replace/torch_wl_kernel.py

```diff
@@ -95,61 +72,33 @@ def forward(
             return out
 
         if x1_is_x2:
-            _ixs = x1.flatten().to(torch.int64).tolist()
-            all_graphs = [self.graph_lookup[i] for i in _ixs]
-
-            # No selection requires
+            indices = x1.flatten().to(torch.int64).tolist()
+            all_graphs = indices
             select = None
         else:
-            _ixs1 = x1.flatten().to(torch.int64).tolist()
-            _ixs2 = x2.flatten().to(torch.int64).tolist()
-            all_graphs = [self.graph_lookup[i] for i in _ixs1 + _ixs2]
+            indices1 = x1.flatten().to(torch.int64).tolist()
+            indices2 = x2.flatten().to(torch.int64).tolist()
+            all_graphs = indices1 + indices2
+            select = lambda K: K[: len(indices1), len(indices1):]
+
+        # Handle the special case for -1
+        all_graphs = [
+            len(self.graph_lookup) - 1 if i == -1 else i for i in all_graphs
+        ]
```

Comment:
Can you mention what makes `-1` a special case, not sure from this diff if I can establish what it actually is.